airflow/contrib/hooks/opsgenie_alert_hook.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import json
+
+from airflow.hooks.http_hook import HttpHook
+
+
+class OpsgenieAlertHook(HttpHook):
+    """
+    This hook allows you to post alerts to Opsgenie.
+    Accepts a connection that has an Opsgenie API key as the connection's password.
+    Each Opsgenie API key can be pre-configured to a team integration.
+    You can override these defaults in this hook.
+
+    :param http_conn_id: Http connection ID with host as "https://api.opsgenie.com/"
+                      and Opsgenie API key as the connection's password
+                      (e.g. "eb243592-faa2-4ba2-a551q-1afdf565c889")
+    :type http_conn_id: str
+    :param payload: Opsgenie API Create Alert payload values
+                    See https://docs.opsgenie.com/docs/alert-api#section-create-alert
+    :type payload: dict
+    :param proxy: Proxy to use to make the Opsgenie Alert API call
+    :type proxy: str
+    """
+    def __init__(self,
+                 http_conn_id,
+                 payload={},
+                 proxy=None,
+                 *args,
+                 **kwargs
+                 ):
+        super(OpsgenieAlertHook, self).__init__(*args, **kwargs)
+        self.http_conn_id = http_conn_id
+        self.payload = payload
+        self.proxy = proxy
+
+    def execute(self):
+        """
+        Remote Popen (actually execute the Opsgenie Alert call)
+        """
+        proxies = {}
+        if self.proxy:
+            # we only need https proxy for Opsgenie, as the endpoint is https
+            proxies = {'https': self.proxy}
+        api_key = self.get_connection(self.http_conn_id).password
+        return self.run(endpoint='v2/alerts',
+                        data=json.dumps(self.payload, sort_keys=True),
```

Comment:
Maybe we should remove `sort_keys`? I personally think is unnecessary.

Reply:
It's needed to ensure the `test_opsgenie_alert_hook` passes consistently.

The test is `self.assertEqual(resp.request.body, json.dumps(self._expected_payload_dict, sort_keys=True))` and before I added `sort_keys` to both, since it's a string assertion, it would fail dependent on how the keys were loaded.

Reply:
Why not do `self.assertEqual(json.loads(resp.request.body), self._expected_payload_dict)` instead?

Reply:
I believe I originally tried that and had inconsistent failures, but I made the suggested change, and hopefully I was mistaken.